### PR TITLE
Alias teleport caller only if it is a contract

### PR DESCRIPTION
**Pull request by godzillaba**

### CODE REVIEW

File: script/0_DeployL2Contracts.s.sol [hunk 1]
```diff
@@ -15,7 +15,6 @@ contract DeployL2Contracts is Script {
 
         vm.startBroadcast(deployerPrivateKey);
 
-        // @review - this is not deployed using a CREATE2 factory as described in README
```
Comment (godzillaba) [inline script/0_DeployL2Contracts.s.sol:18]: foundry scripts use `0x4e59b44847b379578588920ca78fbf26c0b4956c` when you use this syntax

Comment (gzeoneth) [reply script/0_DeployL2Contracts.s.sol:18]: I see, can you add a comment?

File: contracts/L1Teleporter.sol [hunk 2]
```diff
@@ -234,4 +234,9 @@ contract L1Teleporter is Pausable, AccessControl, L2ForwarderPredictor, IL1Telep
         results.l2l3TokenBridgeCost =
             gasParams.l2l3TokenBridgeMaxSubmissionCost + gasParams.l2l3TokenBridgeGasLimit * gasParams.l3GasPriceBid;
     }
+
+    /// @dev Alias the address if it is a contract
```
Comment (gzeoneth) [inline contracts/L1Teleporter.sol:238]: also note that it does not work in certain edge case e.g. within constructor

Comment (godzillaba) [reply contracts/L1Teleporter.sol]: added a few comments

**Review by gzeoneth — APPROVED** [review]
LGTM

**Pull request merged**